src/grants/GrantCheck.ts

```diff
@@ -0,0 +1,176 @@
+import { Logger } from "matrix-appservice-bridge";
+import { Appservice, Intent, MatrixError } from "matrix-bot-sdk";
+import { BridgeConfig, BridgeConfigGitLab, BridgePermissionLevel } from "../Config/Config";
+import { GitHubRepoConnection, GitLabRepoConnection } from "../Connections";
```

Comment:
Dropping into a thread (@justinbot):

> (2) adding more connection-specific code outside of connections. Hopefully both can be avoided with a bit of refactoring

This was a hard choice. The problem I was facing is that I didn't want to repeat that logic N times for every GitHub connection type, so instead I pasted it inside the grant code. We could probably stop cheating and create subclasses of GrantChecker and not store them alongside the grant checker itself.